src/discord/events/message/registerTeam.ts

```diff
@@ -0,0 +1,214 @@
+import Discord from 'discord.js';
+
+import { Config } from '../../../entities/config';
+import logger from '../../../logger';
+import { colors } from '../../constants';
+import { Team } from '../../../entities/team';
+
+class TeamInfo {
+  idArr: string[] = [];
+
+  name = '';
+
+  description = '';
+
+  table = 0;
+
+  user = '';
+
+  constructor(i: string) {
+    this.user = i;
+  }
+
+  addMembers(mem: string[]): void {
+    this.idArr = mem;
+  }
+
+  addUser(): void {
+    this.idArr.push(this.user);
+  }
+
+  addName(n: string): void {
+    this.name = n;
+  }
+
+  addDescription(d: string): void {
+    this.description = d;
+  }
+
+  addTable(t: number): void {
+    this.table = t;
+  }
+}
+
+let registrationInProgress = false;
+let decrement = false;
+let findName: Team;
+let findTable: Team;
```

Comment:
Because all these values exist at the root scope of this file, they'll be overridden as other users enter into this flow. For example, if 5 users use `!registerTeam` simultaneously, these values will bounce around and lose their correctness. 

We need to create a map of these items to a given user's id to track them through this flow.